solidity-plugin/src/main/java/org/sonarsource/solidity/SoliditySensor.java

```diff
@@ -82,25 +83,52 @@ public void analyzeFiles(SensorContext context, List<InputFile> files) {
     for (InputFile file : files) {
       String lastAnalyzedFile = file.toString();
       if (inSonarQube(context)) {
-        LOG.debug("Analyzing: " + lastAnalyzedFile);
-        getSyntaxHighlighting(context, file).save();
+        try {
+          LOG.debug("Analyzing: " + lastAnalyzedFile);
+          SolidityParser parser = Utils.returnParserUnitFromParsedFile(file.contents());
+          getSyntaxHighlighting(parser, context, file).save();
+          saveFileMeasures(context, computeMeasures(parser, fileLinesContextFactory.createFor(file), file), file);
+        } catch (IOException e) {
+
+        }
       } else {
         LOG.debug(lastAnalyzedFile);
       }
     }
   }
 
-  public NewHighlighting getSyntaxHighlighting(SensorContext context, InputFile inputFile) {
-    this.highlighting = context.newHighlighting().onFile(inputFile);
-    try {
-      SolidityParser parser = Utils.returnParserUnitFromParsedFile(inputFile.contents());
-      MyVisitor visitor = new MyVisitor(this.highlighting);
-      visitor.visitTokens(parser.getTokenStream());
-      visitor.highlightComments(parser);
-    } catch (IOException e) {
-      LOG.error(e.toString());
-    }
-    return this.highlighting;
+  private FileMeasures computeMeasures(SolidityParser parser, FileLinesContext fileLinesContext, InputFile file) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove these unused method parameters. ([squid:S1172](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=16&issues=AWQTGK75bXXTsx7O4tIU&open=AWQTGK75bXXTsx7O4tIU)

[ISSUE_KEY]: AWQTGK75bXXTsx7O4tIU